framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -923,6 +1066,50 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
                 "ValueAnnotatedTypeFactory: unexpected class: " + first.getClass());
     }
 
+    public AnnotationMirror createIntRangeAnnotation(Range range) {
+        if (range.from > range.to || range.from == Long.MIN_VALUE && range.to == Long.MAX_VALUE) {
+            return UNKNOWNVAL;
+        } else if (!range.isWiderThan(MAX_VALUES)) {
+            List<Long> values = new ArrayList<>();
+            for (long value = range.from; value <= range.to; value++) {
+                values.add(value);
+            }
+            return createIntValAnnotation(values);
+        } else {
+            AnnotationBuilder builder = new AnnotationBuilder(processingEnv, IntRange.class);
+            builder.setValue("from", range.from);
+            builder.setValue("to", range.to);
+            return builder.build();
+        }
+    }
+
+    public static Range getRangeFromValues(List<Long> values) {
```

Comment:
Could you please add a Javadoc comment to the methods from here down in the file -- and to any method that you defined?